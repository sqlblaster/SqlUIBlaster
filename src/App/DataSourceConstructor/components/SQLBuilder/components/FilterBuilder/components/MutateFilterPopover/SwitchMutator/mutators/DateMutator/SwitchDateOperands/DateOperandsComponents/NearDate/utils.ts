import { PluralDateComponent } from './plural-date-components';

export const getThisDateComponentNotation = (
  dateComponent: PluralDateComponent
) => {
  const singularForm = dateComponent.substr(0, dateComponent.length - 1);

  const thisDateNotation =
    singularForm === 'Day' ? 'Today' : `This ${singularForm}`;

  return thisDateNotation.toLowerCase();
};
