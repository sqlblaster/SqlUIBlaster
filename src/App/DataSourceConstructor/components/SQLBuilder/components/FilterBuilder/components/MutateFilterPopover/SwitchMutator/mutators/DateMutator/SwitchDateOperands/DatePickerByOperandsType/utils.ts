import { SingleDateOperands } from '../DateOperandsComponents/SingleDate/model';
import { FixedDateOperands } from './components/FixedDatePicker/model';
import { RelativeDateOperands } from './components/RelativeDatePicker/model';
import { VariableDateOperands } from './components/VariableDatePicker/model';
import { DateType } from './date-operand-types';

export const getNewOperands = (dateType: DateType): SingleDateOperands => {
  switch (dateType) {
    case 'Fixed date':
      return new FixedDateOperands();
    case 'Relative date':
      return new RelativeDateOperands();
    case 'Variable date':
      return new VariableDateOperands();
    default:
      throw new Error('unknown date type is provided');
  }
};
