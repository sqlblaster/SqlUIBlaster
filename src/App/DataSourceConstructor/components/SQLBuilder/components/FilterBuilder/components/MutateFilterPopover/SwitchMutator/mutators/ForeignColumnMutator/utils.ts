import { ForeignColumnMutatorProps } from './props';

export const getModels = (
  props: Pick<ForeignColumnMutatorProps, 'enums' | 'column'>
): string[] | undefined => {
  const {
    enums,
    column: {
      lastColumn: { foreignModelName, modelName, type }
    }
  } = props;

  return type === 'uuid' && !enums
    ? undefined
    : type === 'uuid' && enums
    ? enums
    : type === 'primary'
    ? [modelName]
    : foreignModelName
    ? [foreignModelName]
    : undefined;
};
