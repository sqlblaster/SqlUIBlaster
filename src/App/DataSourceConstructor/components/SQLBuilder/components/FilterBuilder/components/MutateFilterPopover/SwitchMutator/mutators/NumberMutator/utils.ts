import { NumberOperands } from './NumberOperation';
import { NumberOperator } from './operators';

export const getDefaultOperands = (
  operator: NumberOperator
): NumberOperands => {
  if (operator === 'Between') {
    return [0, 0];
  } else if (operator === 'Is empty' || operator === 'Is not empty') {
    return undefined;
  } else {
    return 0;
  }
};
