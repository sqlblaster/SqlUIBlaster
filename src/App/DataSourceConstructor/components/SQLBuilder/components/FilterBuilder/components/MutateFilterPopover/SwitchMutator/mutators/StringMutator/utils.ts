import { StringOperator } from './operators';

export const hasCaseSensivity = (operator: StringOperator): boolean => {
  switch (operator) {
    case 'Contains':
    case 'Does not contain':
    case 'Starts with':
    case 'Ends with':
      return true;
    default:
      return false;
  }
};
