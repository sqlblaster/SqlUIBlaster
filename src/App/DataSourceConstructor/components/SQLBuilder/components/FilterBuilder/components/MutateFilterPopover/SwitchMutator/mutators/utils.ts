import {
  Column,
  Table
} from 'src/App/DataSourceConstructor/schemas';
import { ColumnBranch } from '../../../../../common/ColumnSelector/ColumnItem/SelectedColumn.models';

export const getEnums = (
  tables: Table[],
  columnBranch: ColumnBranch
): Column['enums'] => {
  const { tableName, columnName } = columnBranch;
  let enums: Column['enums'];

  const table = tables.find(t => t.name === tableName);
  let column: Column | undefined;
  table && (column = table.columns.find(c => c.name === columnName));
  column && (enums = column.enums);

  return enums;
};
