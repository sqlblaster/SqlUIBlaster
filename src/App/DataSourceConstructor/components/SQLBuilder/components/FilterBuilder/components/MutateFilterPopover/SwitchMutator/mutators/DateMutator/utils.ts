import { DateOperation } from './DateOperation';
import { CoupleDateOperation } from './SwitchDateOperands/DateOperandsComponents/CoupleDates/model';
import { EmptyDateOperation } from './SwitchDateOperands/DateOperandsComponents/EmptyDates/model';
import { NearDateOperation } from './SwitchDateOperands/DateOperandsComponents/NearDate/model';
import { SingleDateOperation } from './SwitchDateOperands/DateOperandsComponents/SingleDate/model';

export const getDateOperationVia = (
  operator: DateOperation['operator']
): DateOperation => {
  switch (operator) {
    case 'Previous':
    case 'Next':
      return new NearDateOperation({ operator });
    case 'Before':
    case 'After':
    case 'On':
      return new SingleDateOperation({ operator });
    case 'Between':
      return new CoupleDateOperation({ operator });
    case 'Is empty':
    case 'Is not empty':
    default:
      return new EmptyDateOperation({ operator });
  }
};
