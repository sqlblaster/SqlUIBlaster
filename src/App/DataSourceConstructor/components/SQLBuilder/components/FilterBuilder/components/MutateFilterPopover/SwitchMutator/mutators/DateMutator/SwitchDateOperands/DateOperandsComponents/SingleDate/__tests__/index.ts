import { checkEverythingFor } from './utils/check-everything-for';

describe('Single Date: Before, After, On operations', () => {
  checkEverythingFor('Before');
  checkEverythingFor('After');
  checkEverythingFor('On');
});
