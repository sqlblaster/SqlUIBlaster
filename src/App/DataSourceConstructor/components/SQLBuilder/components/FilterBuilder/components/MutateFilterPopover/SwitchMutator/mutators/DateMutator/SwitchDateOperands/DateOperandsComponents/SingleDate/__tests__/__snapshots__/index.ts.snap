// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Date: Before, After, On operations should render sql queries properly on 'After' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'After' operator - Fixed date - When enabled time is disabled 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp >= ('2019-01-01 00:00:00'::timestamp + interval '1 day')::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'After' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'After' operator - Fixed date - With date only 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp >= ('2019-01-01 00:00:00'::timestamp + interval '1 day')::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'After' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'After' operator - Relative date - Beginning of the current month 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp >= (date_trunc('month', now()) + interval '1 day')::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'After' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'After' operator - Relative date - Beginning of the current quarter 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp >= (date_trunc('quarter', now()) + interval '1 day')::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'After' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'After' operator - Relative date - Beginning of the current week 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp >= (date_trunc('week', now()) + interval '1 day')::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'After' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'After' operator - Relative date - Beginning of the current year 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp >= (date_trunc('year', now()) + interval '1 day')::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'After' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'After' operator - Relative date - End of the current month 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp >= ((date_trunc('month', now()) + interval '1 month - 1 day') + interval '1 day')::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'After' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'After' operator - Relative date - End of the current quarter 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp >= ((date_trunc('quarter', now()) + interval '3 month - 1 day') + interval '1 day')::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'After' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'After' operator - Relative date - End of the current week 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp >= ((date_trunc('week', now()) + interval '6 days') + interval '1 day')::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'After' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'After' operator - Relative date - End of the current year 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp >= ((date_trunc('year', now()) + interval '1 year - 1 day') + interval '1 day')::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'After' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'After' operator - Relative default date - Today 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp >= (now()::date + interval '1 day')::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'After' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'After' operator - Should pick one of variables and save 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp >= ($1::timestamp + interval '1 day')::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'After' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: After' operator - Fixed date - With certain time 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp > '2019-01-01 06:00:00'::timestamp
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'After' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: After' operator - Fixed date - With date and default time 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp > '2019-01-01 00:00:00'::timestamp
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'Before' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'Before' operator - Fixed date - When enabled time is disabled 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp < ('2019-01-01 00:00:00'::timestamp)::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'Before' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'Before' operator - Fixed date - With date only 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp < ('2019-01-01 00:00:00'::timestamp)::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'Before' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'Before' operator - Relative date - Beginning of the current month 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp < (date_trunc('month', now()))::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'Before' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'Before' operator - Relative date - Beginning of the current quarter 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp < (date_trunc('quarter', now()))::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'Before' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'Before' operator - Relative date - Beginning of the current week 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp < (date_trunc('week', now()))::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'Before' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'Before' operator - Relative date - Beginning of the current year 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp < (date_trunc('year', now()))::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'Before' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'Before' operator - Relative date - End of the current month 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp < ((date_trunc('month', now()) + interval '1 month - 1 day'))::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'Before' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'Before' operator - Relative date - End of the current quarter 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp < ((date_trunc('quarter', now()) + interval '3 month - 1 day'))::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'Before' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'Before' operator - Relative date - End of the current week 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp < ((date_trunc('week', now()) + interval '6 days'))::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'Before' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'Before' operator - Relative date - End of the current year 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp < ((date_trunc('year', now()) + interval '1 year - 1 day'))::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'Before' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'Before' operator - Relative default date - Today 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp < (now()::date)::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'Before' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'Before' operator - Should pick one of variables and save 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp < ($1::timestamp)::date
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'Before' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: Before' operator - Fixed date - With certain time 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp < '2019-01-01 06:00:00'::timestamp
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'Before' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: Before' operator - Fixed date - With date and default time 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Created At\\"::timestamp < '2019-01-01 00:00:00'::timestamp
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'On' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'On' operator - Fixed date - When enabled time is disabled 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Days', (\\"orders1\\".\\"Created At\\"::timestamp)::timestamp) = date_trunc('Days', ('2019-01-01 00:00:00'::timestamp)::timestamp)
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'On' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'On' operator - Fixed date - With date only 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Days', (\\"orders1\\".\\"Created At\\"::timestamp)::timestamp) = date_trunc('Days', ('2019-01-01 00:00:00'::timestamp)::timestamp)
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'On' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'On' operator - Relative date - Beginning of the current month 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Days', (\\"orders1\\".\\"Created At\\"::timestamp)::timestamp) = date_trunc('Days', (date_trunc('month', now()))::timestamp)
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'On' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'On' operator - Relative date - Beginning of the current quarter 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Days', (\\"orders1\\".\\"Created At\\"::timestamp)::timestamp) = date_trunc('Days', (date_trunc('quarter', now()))::timestamp)
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'On' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'On' operator - Relative date - Beginning of the current week 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Days', (\\"orders1\\".\\"Created At\\"::timestamp)::timestamp) = date_trunc('Days', (date_trunc('week', now()))::timestamp)
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'On' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'On' operator - Relative date - Beginning of the current year 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Days', (\\"orders1\\".\\"Created At\\"::timestamp)::timestamp) = date_trunc('Days', (date_trunc('year', now()))::timestamp)
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'On' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'On' operator - Relative date - End of the current month 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Days', (\\"orders1\\".\\"Created At\\"::timestamp)::timestamp) = date_trunc('Days', ((date_trunc('month', now()) + interval '1 month - 1 day'))::timestamp)
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'On' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'On' operator - Relative date - End of the current quarter 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Days', (\\"orders1\\".\\"Created At\\"::timestamp)::timestamp) = date_trunc('Days', ((date_trunc('quarter', now()) + interval '3 month - 1 day'))::timestamp)
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'On' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'On' operator - Relative date - End of the current week 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Days', (\\"orders1\\".\\"Created At\\"::timestamp)::timestamp) = date_trunc('Days', ((date_trunc('week', now()) + interval '6 days'))::timestamp)
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'On' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'On' operator - Relative date - End of the current year 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Days', (\\"orders1\\".\\"Created At\\"::timestamp)::timestamp) = date_trunc('Days', ((date_trunc('year', now()) + interval '1 year - 1 day'))::timestamp)
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'On' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'On' operator - Relative default date - Today 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Days', (\\"orders1\\".\\"Created At\\"::timestamp)::timestamp) = date_trunc('Days', (now()::date)::timestamp)
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'On' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: 'On' operator - Should pick one of variables and save 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Days', (\\"orders1\\".\\"Created At\\"::timestamp)::timestamp) = date_trunc('Days', ($1::timestamp)::timestamp)
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'On' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: On' operator - Fixed date - With certain time 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Minutes', (\\"orders1\\".\\"Created At\\"::timestamp)::timestamp) = date_trunc('Minutes', ('2019-01-01 06:00:00'::timestamp)::timestamp)
LIMIT 2000
"
`;

exports[`Single Date: Before, After, On operations should render sql queries properly on 'On' operator:
    on all cases:
      - Fixed date:
        - With date only
        - With date and time
        - Enabled time to be disabled
      - Variable date:
        - When there are no variables should show appropriate text and can't save
        - When there are variables should show placeholder and can't save
        - Should pick one of variables and save
      - All Relative date cases: On' operator - Fixed date - With date and default time 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Minutes', (\\"orders1\\".\\"Created At\\"::timestamp)::timestamp) = date_trunc('Minutes', ('2019-01-01 00:00:00'::timestamp)::timestamp)
LIMIT 2000
"
`;
